### PR TITLE
Fixes #370: acli push:files creates an extra "files" directory.

### LINKED ISSUE #370: acli push:files creates an extra "files" directory
Note this is happening both with Acquia CLI 1.1.1 that is currently installed in IDEs, and HEAD, which I tested with and observed the same behavior.

```
ide:~/project (master) $ stat docroot/sites/default/files/chocolate.jpg 
  File: 'docroot/sites/default/files/chocolate.jpg'
  Size: 88261           Blocks: 176        IO Block: 4096   regular file
Device: 10303h/66307d   Inode: 3678750     Links: 1
Access: (0644/-rw-r--r--)  Uid: (10000/     ide)   Gid: ( 1002/     ide)
Access: 2020-12-02 03:23:44.000000000 -0500
Modify: 2019-02-15 08:41:03.000000000 -0500
Change: 2020-12-02 03:23:50.089588151 -0500
 Birth: -
ide:~/project (master) $ acli push:files -vvv

Box Requirements Checker
========================

> Using PHP 7.3.21
> PHP is using the following php.ini file:
  /usr/local/php7.3/etc/cli/php.ini

> Checking Box requirements:
  ✔ The application requires the version "^7.3" or greater.
  ✔ The application requires the extension "json".
  ✔ The package "brick/math" requires the extension "json".
  ✔ The package "guzzlehttp/guzzle" requires the extension "json".
  ✔ The package "ramsey/uuid" requires the extension "json".
  ✔ The package "composer/ca-bundle" requires the extension "openssl".
  ✔ The package "padraic/humbug_get_contents" requires the extension "openssl".
  ✔ The package "composer/ca-bundle" requires the extension "pcre".
  
                                                                                
 [OK] Your system is ready to run the application.                              
                                                                                

[debug] Could not determine if Acquia CLI has a new version available.
Using Cloud Application <REDACTED_SITEGROUP>

 Choose a Cloud Platform environment:
  [0] Dev, dev (vcs: pipelines-build-master)
  [1] Stage, test (vcs: tags/2020-10-19)
 > 1

[debug] Using environment Stage <REDACTED_UUID>
 Overwrite the public files directory on test with a copy of the files from the current machine? (yes/no) [yes]:
 > y

    ⌛ Pushing public files directory to remote machine...
sending incremental file list
files/
files/chocolate.jpg
         88.26K 100%   52.92MB/s    0:00:00 (xfr#1, ir-chk=1209/1222)

sent 232.75K bytes  received 551 bytes  93.32K bytes/sec
    ✔ Pushing public files directory to remote machine
```

And on the Cloud Platform server:

```
<REDACTED_SITEGROUP>@staging-14821:/var/www/html/<REDACTED_SITEGROUP>.test$ stat /mnt/gfs/<REDACTED_SITEGROUP>.test/sites/default/files/chocolate.jpg
stat: cannot stat '/mnt/gfs/anavarred8.test/sites/default/files/chocolate.jpg': No such file or directory
<REDACTED_SITEGROUP>@staging-14821:/var/www/html/<REDACTED_SITEGROUP>.test$ stat /mnt/gfs/<REDACTED_SITEGROUP>.test/sites/default/files/files/chocolate.jpg
  File: '/mnt/gfs/<REDACTED_SITEGROUP>.test/sites/default/files/files/chocolate.jpg'
  Size: 88261           Blocks: 176        IO Block: 131072 regular file
Device: 30h/48d Inode: 1879097523  Links: 1
Access: (0644/-rw-r--r--)  Uid: (28243/<REDACTED_SITEGROUP>)   Gid: (28243/<REDACTED_SITEGROUP>)
Access: 2020-12-02 08:24:42.371876000 +0000
Modify: 2019-02-15 13:41:03.000000000 +0000
Change: 2020-12-02 08:24:42.375876440 +0000
 Birth: -
```

Note the extra `files` directory.

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

File: tests/phpunit/src/Commands/Push/PushFilesCommandTest.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ protected function mockExecuteAcsfRsync(
       'rsync',
       '-rltDvPhe',
       'ssh -o StrictHostKeyChecking=no',
-      $this->projectFixtureDir . '/docroot/sites/default/',
+      $this->projectFixtureDir . '/docroot/sites/default/files',
```
Comment (anavarre) [inline tests/phpunit/src/Commands/Push/PushFilesCommandTest.php:147]: Are you not missing the trailing slash here?

**Comment by grasmash** [comment]
This is failing for reasons unrelated to the changes. The Windows build is defaulting to PHP 8.0.0, and the PHP 7.4 build is passing tests but failing to execute coveralls.

**Comment by anavarre** [comment]
FYI you're not working from a branch but from `master`.

**Comment by anavarre** [comment]
Not sure if it's me or the PR but I'm running into an issue when trying to test this PR

```
$ /tmp/cli/bin/acli push:files -vvv
Acquia CLI version: @package_version@

In AbstractProvider.php line 539:
                                                                       
  [UnexpectedValueException]                                           
  Invalid response received from Authorization Server. Expected JSON.  
                                                                       

Exception trace:
  at /tmp/cli/vendor/league/oauth2-client/src/Provider/AbstractProvider.php:539
 League\OAuth2\Client\Provider\AbstractProvider->getAccessToken() at /tmp/cli/vendor/typhonius/acquia-php-sdk-v2/src/Connector/Connector.php:74
 AcquiaCloudApi\Connector\Connector->AcquiaCloudApi\Connector\{closure}() at /tmp/cli/vendor/symfony/cache/LockRegistry.php:99
 Symfony\Component\Cache\LockRegistry::compute() at /tmp/cli/vendor/symfony/cache/Traits/ContractsTrait.php:88
 Symfony\Component\Cache\Adapter\AbstractAdapter->Symfony\Component\Cache\Traits\{closure}() at /tmp/cli/vendor/symfony/cache-contracts/CacheTrait.php:70
 Symfony\Component\Cache\Adapter\AbstractAdapter->doGet() at /tmp/cli/vendor/symfony/cache/Traits/ContractsTrait.php:95
 Symfony\Component\Cache\Adapter\AbstractAdapter->doGet() at /tmp/cli/vendor/symfony/cache-contracts/CacheTrait.php:33
 Symfony\Component\Cache\Adapter\AbstractAdapter->get() at /tmp/cli/vendor/typhonius/acquia-php-sdk-v2/src/Connector/Connector.php:75
 AcquiaCloudApi\Connector\Connector->createRequest() at /tmp/cli/vendor/typhonius/acquia-php-sdk-v2/src/Connector/Connector.php:92
 AcquiaCloudApi\Connector\Connector->sendRequest() at /tmp/cli/vendor/typhonius/acquia-php-sdk-v2/src/Connector/Client.php:128
 AcquiaCloudApi\Connector\Client->makeRequest() at /tmp/cli/vendor/typhonius/acquia-php-sdk-v2/src/Connector/Client.php:117
 AcquiaCloudApi\Connector\Client->request() at /tmp/cli/vendor/typhonius/acquia-php-sdk-v2/src/Endpoints/Applications.php:40
 AcquiaCloudApi\Endpoints\Applications->get() at /tmp/cli/src/Command/CommandBase.php:874
 Acquia\Cli\Command\CommandBase->getCloudApplication() at /tmp/cli/src/Command/CommandBase.php:695
 Acquia\Cli\Command\CommandBase->determineCloudApplication() at /tmp/cli/src/Command/Pull/PullCommandBase.php:571
 Acquia\Cli\Command\Pull\PullCommandBase->determineEnvironment() at /tmp/cli/src/Command/Push/PushFilesCommand.php:45
 Acquia\Cli\Command\Push\PushFilesCommand->execute() at /tmp/cli/vendor/symfony/console/Command/Command.php:258
 Symfony\Component\Console\Command\Command->run() at /tmp/cli/src/Command/CommandBase.php:354
 Acquia\Cli\Command\CommandBase->run() at /tmp/cli/vendor/symfony/console/Application.php:938
 Symfony\Component\Console\Application->doRunCommand() at /tmp/cli/vendor/symfony/console/Application.php:266
 Symfony\Component\Console\Application->doRun() at /tmp/cli/vendor/symfony/console/Application.php:142
 Symfony\Component\Console\Application->run() at /tmp/cli/bin/acli:89

push:files [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [--] <command> [<dir> [<environmentId>]]
```

I did run the `ckc` command and ran `composer install`. Anything I missed?

**Review by danepowell — APPROVED** [review]

**Comment by grasmash** [comment]
> FYI you're not working from a branch but from `master`.

Not sure what you mean... this is from the `issue-370-files-dir` branch.

**Comment by anavarre** [comment]
Sorry, meant that you're not working from your fork.

**Comment by grasmash** [comment]
```
touch docroot/sites/default/files/testing-371
./bin/acli push:files -vvv
acli ssh [site.env] -- ls -lash docroot/sites/default/files/ | grep 371
```

Seems to work for me. :)

**Comment by grasmash** [comment]
> Sorry, meant that you're not working from your fork.

Yeah my fork is messed up.

**Pull request merged**